### PR TITLE
Include WHO reference strains, vaccine strains, NCBI genotypes in trees

### LINKED ISSUE #24: Primary key shouldn't include version number in GenBank accession
### Current Behavior
Currently, the primary key that's associated with each sample during ingest includes GenBank version ID, eg `.1` resulting in metadata like:
```
accession
Z80832.1
Z80831.1
```
and sequences FASTA like:
```
>Z80832.1
ATG...
```

This necessitates annotations like what I used for PR #23 like:
```
AF266288.2	strain	Measles strain Edmonston WT
AF266288.2	date	1954
AF266288.2	region	North America
AF266288.2	country	USA
```

However, authors may often update records in GenBank. This would cause the ingested accession to increment and would disassociate our annotation match.

I also think that the `.1`, `.2`, etc... adds unnecessary noise when looking at fasta and metadata. GenBank accessions for mpox and SARS-CoV-2 don't include the version ID.

### Expected behavior
Drop version ID from GenBank ingest.

### PR DESCRIPTION
**Pull request by kimandrews**
## Description of proposed changes
This PR adds specific sequences and metadata to the measles trees, including WHO reference sequences, vaccine strains, and genotypes reported on NCBI, through the following steps:
*  Dropping GenBank accession version number (resolves #24)
* Force-include vaccine strains in N450 and genome trees (follows up on https://github.com/nextstrain/measles/pull/23)
* Add WHO genotype reference strains to N450 tree (these sequences are too short to include in the genome tree)
* Use a custom script to parse genotypes from the `virus-name` field of the NDJSON record from NCBI (addresses https://github.com/nextstrain/measles/issues/16)
* Include genotype coloring on N450 and genome trees

## Related issue(s)
https://github.com/nextstrain/measles/pull/23
#24
https://github.com/nextstrain/pathogen-repo-guide/pull/40
https://github.com/nextstrain/measles/issues/16

## Checklist

- [ ] Checks pass

### CODE REVIEW

File: ingest/rules/curate.smk [hunk 1]
```diff
@@ -105,6 +105,7 @@ rule curate:
                 --abbr-authors-field {params.abbr_authors_field} \
             | ./vendored/apply-geolocation-rules \
                 --geolocation-rules {input.all_geolocation_rules} \
+            | ./bin/parse-measles-genotype-names.py \
```
Comment (joverlee521) [inline ingest/rules/curate.smk:108]: I recommend including the `--genotype-field` flag here and add a param to the config.yaml. 
If the `virus_name` field ever gets renamed or if we decide to switch to a new field to get the genotype, we can just edit the config.yaml.

**Review by joverlee521 — APPROVED** [review]
It's good to see the NCBI genotypes parsing works for ~70% of the current data! 
I've left minor comments, but none of them are blocking especially since this work is necessary for your Nextclade dataset!

**Review by jameshadfield — APPROVED** [review]

File: ingest/bin/parse-measles-genotype-names.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    return (
+        genotype_name)
+
+def main():
+    args = parse_args()
+
+    for index, record in enumerate(stdin):
+        record = json.loads(record)
+        record['genotype_ncbi'] = record[args.genotype_field][:]
```
[the remark this answers is not in the document]
Comment (kimandrews) [reply ingest/bin/parse-measles-genotype-names.py]: Done in cce8b3c

File: ingest/bin/parse-measles-genotype-names.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        record['genotype_ncbi'] = _set_genotype_name(record)
+        allowed_values = ['A', 'B1', 'B2', 'B3', 'C1', 'C2', 'D1', 'D2', 'D3', 'D4', 'D5', 'D6', 'D7', 'D8', 'D9', 'D10', 'D11', 'E', 'F', 'G1', 'G2', 'G3', 'H1', 'H2']
```
[the remark this answers is not in the document]
Comment (kimandrews) [reply ingest/bin/parse-measles-genotype-names.py]: Done in cce8b3c

File: ingest/rules/curate.smk (hunk 1 above)
Comment (kimandrews) [reply ingest/rules/curate.smk]: Done in cce8b3c

**Review by joverlee521 — APPROVED** [review]
Thanks for address my comments, the latest changes look good!

**Pull request merged**